src/probability/cond_count.lean

```diff
@@ -0,0 +1,194 @@
+
+## main definition
+
+* `probability_theory.cond_count`: given a set `s`, `cond_count s` is the counting measure
+  conditioned on `s`. This is a probability measure when `s` is finite and nonempty.
+
+## notes
+
+The original aim of this file is to provide a measure theoretic method of describing the
+probability an element of a set `s` satisfies some predicate `P`. Our current formulation still
+allow us to describe this by abusing the definitional equality of sets and predicates by simply
+writing `cond_count s P`. We should avoid this however as none of the lemmas are written for
+predicates.
+-/
+
+noncomputable theory
+
+open_locale probability_theory
+
+open measure_theory measurable_space
+
+namespace probability_theory
+
+variables {α : Type*} [measurable_space α]
+
+/-- Given a set `s`, `cond_count s` is the counting measure conditioned on `s`. In particular,
+`cond_count s t` is the proportion of `s` that is contained in `t`.
+
+This is a probability measure when `s` is finite and nonempty and is given by
+`probability_theory.cond_count_is_probability_measure`. -/
+def cond_count (s : set α) : measure α := measure.count[|s]
+
+@[simp] lemma cond_count_empty_meas : (cond_count ∅ : measure α) = 0 :=
+by simp [cond_count]
+
+lemma cond_count_empty {s : set α} : cond_count s ∅ = 0 :=
+by simp
+
+variables [measurable_singleton_class α]
+
+lemma cond_count_is_probability_measure {s : set α} (hs : s.finite) (hs' : s.nonempty) :
+  is_probability_measure (cond_count s) :=
+{ measure_univ :=
+  begin
+    rw [cond_count, cond_apply _ hs.measurable_set, set.inter_univ, ennreal.inv_mul_cancel],
+    { exact λ h, hs'.ne_empty $ measure.empty_of_count_eq_zero h },
+    { exact (measure.count_apply_lt_top.2 hs).ne }
+  end }
+
+lemma cond_count_singleton (a : α) (t : set α) [decidable (a ∈ t)] :
+  cond_count {a} t = if a ∈ t then 1 else 0 :=
+begin
+  rw [cond_count, cond_apply _ (measurable_set_singleton a), measure.count_singleton,
+    ennreal.inv_one, one_mul],
+  split_ifs,
+  { rw [(by simpa : ({a} : set α) ∩ t = {a}), measure.count_singleton] },
+  { rw [(by simpa : ({a} : set α) ∩ t = ∅), measure.count_empty] },
+end
+
+variables {s t u : set α}
+
+lemma cond_count_inter_self (hs : s.finite):
+  cond_count s (s ∩ t) = cond_count s t :=
+by rw [cond_count, cond_inter_self _ hs.measurable_set]
+
+lemma cond_count_self (hs : s.finite) (hs' : s.nonempty) :
+  cond_count s s = 1 :=
+begin
+  rw [cond_count, cond_apply _ hs.measurable_set, set.inter_self, ennreal.inv_mul_cancel],
+  { exact λ h, hs'.ne_empty $ measure.empty_of_count_eq_zero h },
+  { exact (measure.count_apply_lt_top.2 hs).ne }
+end
+
+lemma cond_count_eq_one_of (hs : s.finite) (hs' : s.nonempty) (ht : s ⊆ t) :
+  cond_count s t = 1 :=
+begin
+  haveI := cond_count_is_probability_measure hs hs',
+  refine eq_of_le_of_not_lt prob_le_one _,
+  rw [not_lt, ← cond_count_self hs hs'],
+  exact measure_mono ht,
+end
+
+lemma pred_true_of_cond_count_eq_one (hs : s.finite)
```

Comment:
Do you need `hs` here? (If `s` is infinite, `h` is impossible, right?)

Reply:
I've changed the requirement to measurability instead

Reply:
I think it should hold even without measurability, no?

Reply:
You are correct. I'll fix after there's cache